### ISSUE TITLE
Want to ignore a Str node in common situation.

### ISSUE THREAD

**Issue by azu** [textlint/textlint-rule-helper#16]
This is a common pattern. We want to provide this detection as a method.

```js
        [Syntax.Str](node){
            if (helper.isChildNode(node, [Syntax.Link, "LinkReference", Syntax.Image, Syntax.BlockQuote, Syntax.Emphasis])) {
                return;
            }
            // logics
        }
```
https://github.com/search?q=LinkReference+textlint&type=code

**Comment by azu** [comment]
Context, We want to add a new node type like `FootnoteReference`, so the helper will support these changes.
https://github.com/textlint/textlint/pull/767

**Comment by azu** [comment]
What the naming of this helper method.

**Comment by azu** [comment]
`helper.isPlainText(Str): boolean` ?

**Comment by azu** [comment]
BlockQuote is an exception.

```
BlockQuote > Paragaph > Str
```

**Comment by azu** [comment]
```js
    isStrUnderParagraph(node: TxtNode): boolean {
        if (node.type !== ASTNodeTypes.Str) {
            return false;
        }
        if (node.parent?.type !== ASTNodeTypes.Paragraph) {
            return false;
        }
        const isInUncontrollableNode = this.isChildNode(node, [ASTNodeTypes.BlockQuote]);
        return !isInUncontrollableNode
    }
```

This is wanted logics.
